runtime/src/merkle_tree.rs

```diff
@@ -0,0 +1,270 @@
+
+use solana_sdk::hash::{Hash, hash, hashv};
+
+#[derive(Debug)]
+pub struct MerkleTree {
```

Comment:
Any reason to go with our own implementation instead of using an existing crate?

Reply:
A few. :)

Anatoly didn't seem particularly keen on adding dependencies and also was interested in exploring parallelization options down the road.  The available crates were a bit of a mine field in terms of one or more of a) their own dependencies, b) maintainer activity, c) implementation WRT parallelization options.  Plus, it was pretty easy!